pyproject.toml

```
@@ -18,7 +18,8 @@ dependencies = [
     'asdf-transform-schemas >=0.2.2',
     'importlib_resources >=3; python_version <"3.9"',
     'jmespath >=0.6.2',
-    'jsonschema >=4.0.1, <4.10.0',
+    #'jsonschema >=4.0.1, <4.10.0',
+    'jsonschema >=4.0.1',
```

```suggestion
    'jsonschema >=4.0.1, <4.10.0',
```

I would limit this PR to fixing the failing tests for `jsonschema 4.10.0+` not unpinning it. As part of that I want it tested on earlier versions in addition to the latest `jsonschema`. As such I request we not change the pin and instead add `jsonschema` to the [`requirements-dev.txt`](https://github.com/asdf-format/asdf/blob/master/requirements-dev.txt) instead. This will test your changes in the later devdeps CI jobs.